app/services/auth/firebase_auth.py

```
@@ -0,0 +1,386 @@
+        session.commit()
+    except (IntegrityError, DataError, OperationalError, SQLAlchemyError) as e:
+        # Rollback database changes.
+        session.rollback()
+        # cleanup orphaned Firebase user
+        try:
+            auth.delete_user(auth_user.uid)
+            logger.error(f"Database commit failed, cleaned up Firebase user: {e}")
+        except Exception as cleanup_error:
+            logger.error(f"Failed to clean up Firebase user after DB failure: {cleanup_error}")
+            logger.error(f"Original DB error: {e}")
+        raise InternalServerError("Failed to create user due to database error and cleanup failure.") from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)